src/crimson/osd/ops_executer.cc

```diff
@@ -459,6 +459,19 @@ auto OpsExecuter::do_write_op(Func&& f, bool um) {
   user_modify = um;
   return std::forward<Func>(f)(pg->get_backend(), obc->obs, txn);
 }
+OpsExecuter::call_errorator::future<> OpsExecuter::do_assert_ver(
+  OSDOp& osd_op,
+  const ObjectState& os)
+{
+  if (!osd_op.op.assert_ver.ver) {
+    return crimson::ct_error::invarg::make();
+  } else if (osd_op.op.assert_ver.ver < os.oi.user_version) {
+    return crimson::ct_error::erange::make();
+  } else if (osd_op.op.assert_ver.ver > os.oi.user_version) {
+    return crimson::ct_error::value_too_large::make();
```

Comment:
i am wondering if we should categorize this case as a `crimson::ct_error::erange` as well.

Reply:
classic osd has the 3 checks, some test cases will check each return error.